packages/vscode-extension/src/VsCodeKieEditorControllerFactory.ts

```diff
@@ -122,14 +128,25 @@ export class VsCodeKieEditorControllerFactory {
             type: mapping.type,
             filePathGlob: mapping.filePathGlob,
             resourcesPathPrefix: this.getWebviewPath(webview, mapping.resourcesPathPrefix),
-            envelopePath: this.getWebviewPath(webview, mapping.envelopePath),
+            envelopeContent: {
+              type: EnvelopeContentType.PATH,
+              path: this.getWebViewPathFromEnvelopeContent(webview, mapping.envelopeContent),
+            },
           })
         );
         return envelopeMappings;
       }, [] as EnvelopeMapping[])
     );
   }
 
+  private getWebViewPathFromEnvelopeContent(webview: Webview, envelopeContent: EnvelopeContent) {
+    if ("path" in envelopeContent) {
```

Comment:
```suggestion
    if (envelopeContent.type === EnvelopeContentType.PATH) {
```
I think it is better to check for the type here.